src/osd/osd_types.cc

```diff
@@ -1928,111 +1928,46 @@ void object_stat_sum_t::encode(bufferlist& bl) const
 void object_stat_sum_t::decode(bufferlist::iterator& bl)
 {
   bool decode_finish = false;
-  DECODE_START_LEGACY_COMPAT_LEN(14, 3, 3, bl);
+  DECODE_START(15, bl);
 #if defined(CEPH_LITTLE_ENDIAN)
-  if (struct_v >= 15) {
-    bl.copy(sizeof(object_stat_sum_t), (char*)(&num_bytes));
-    decode_finish = true;
-  }
+  bl.copy(sizeof(object_stat_sum_t), (char*)(&num_bytes));
+  decode_finish = true;
 #endif
```

Comment:
i'm adding the if() back in so that it's clear how to add new fields next time it happens.